libexternal/src/osxsupport.cpp

```diff
@@ -39,7 +39,7 @@ char *string_to_utf8(const char *p_string)
 	if (s_text_to_unicode_info != NULL)
 	{
 		unsigned int t_string_length;
-		t_string_length = strlen(p_string);
+		t_string_length = (unsigned int)strlen(p_string);
```

Comment:
Actually you may need to explicitly check and fail if `strlen(p_string) > UINDEX_MAX`, I think.